### PR TITLE
Pass headers to authentication programs

### PR DESCRIPTION
**Pull request by stefwalter**
This is for consistency, and so we can pass this data over a JSON message in the future. We'll need encoding over JSON so may as well just use the base64 encoding.

### CODE REVIEW

**Comment by stefwalter** [comment]
@petervo Do you have any further tests or work here?

**Comment by stefwalter** [comment]
Added unit tests and fixed up unit tests. Ready for review and merging.

**Comment by stefwalter** [comment]
Fixed 32-bit issue, and unused function.

File: src/common/cockpitauthorize.c [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+  if (asprintf (&content, "%s:%s", user, password) < 0)
+    {
+      errn = errno;
+      message ("could not build basic response");
+      goto out;
+    }
+
+  clen = strlen (content);
+  elen = cockpit_base64_size (clen);
+  encoded = malloc (elen + 1);
```
Comment (petervo) [inline src/common/cockpitauthorize.c]: Do we need a malloc check here? Or maybe cockpit_base64_ntop should handle null targets?

File: pkg/kubernetes/standalone/src/cockpit-kube-auth/helpers/client.go [hunk 2]
```diff
@@ -228,8 +229,18 @@ func (self *Client) DoRequest(method string, api string, resource string,
...
+	if len(parts) == 0 {
+		return nil, newAuthError("Invalid Authorization line")
+	}
+	authData := ""
+	authType := parts[0]
+	if len(parts) == 2 {
+		authData = parts[1]
+	}
 
+	fmt.Fprintf(os.Stderr, "split: '%s' '%s'\n", authType, authData)
```
Comment (petervo) [inline pkg/kubernetes/standalone/src/cockpit-kube-auth/helpers/client.go]: Uneeded?

File: src/common/cockpitauthorize.c [hunk 3]
```diff
@@ -0,0 +1,577 @@
...
+  size_t elen;
+  int errn = 0;
+
+  if (!input)
+    length = 0;
+
+  if (length > 0)
+    {
+      elen = cockpit_base64_size (length);
+      encoded = malloc (elen);
```
Comment (petervo) [inline src/common/cockpitauthorize.c]: same malloc/cockpit_base64_ntop check

File: src/common/cockpitauthorize.c [hunk 4]
```diff
@@ -0,0 +1,577 @@
...
+      nonce = cockpit_authorize_nonce (nlen);
+      if (!nonce)
+        {
+          errn = errno;
+          message ("could not generate nonce");
+          goto out;
+        }
+
+      clen = cockpit_base64_size (nlen);
+      conv = alloc = malloc (clen);
```
Comment (petervo) [inline src/common/cockpitauthorize.c]: same malloc/cockpit_base64_ntop check

File: src/common/cockpitauthorize.c [hunk 5]
```diff
@@ -0,0 +1,577 @@
...
+      errn = EINVAL;
+      goto out;
+    }
+
+  plen = strlen (prompt);
+  if (plen > 0)
+    {
+      elen = cockpit_base64_size (plen);
+      encoded = malloc (elen);
+      if (cockpit_base64_ntop ((unsigned char *)prompt, plen, encoded, elen) < 0)
```
Comment (petervo) [inline src/common/cockpitauthorize.c]: same malloc/cockpit_base64_ntop check

File: src/common/cockpitauthorize.c (hunk 1 above)
Comment (stefwalter) [inline src/common/cockpitauthorize.c]: Done

File: src/common/cockpitauthorize.c (hunk 3 above)
Comment (stefwalter) [inline src/common/cockpitauthorize.c]: Done.

File: src/common/cockpitauthorize.c (hunk 4 above)
Comment (stefwalter) [inline src/common/cockpitauthorize.c]: Done

File: src/common/cockpitauthorize.c (hunk 5 above)
Comment (stefwalter) [inline src/common/cockpitauthorize.c]: Done

**Comment by stefwalter** [comment]
Fixed for @petervo review.

**Pull request closed without merging**